discovery-provider/src/challenges/challenge.py

```diff
@@ -12,12 +12,17 @@
 def fetch_user_challenges(
     session: Session, challenge_id: str, specifiers: List[str]
 ) -> List[UserChallenge]:
-    return (
+    user_challenges = (
         session.query(UserChallenge).filter(
             UserChallenge.challenge_id == challenge_id,
             UserChallenge.specifier.in_(specifiers),
         )
     ).all()
+    # Re-sort them
+    specifier_map = {
+        user_challenge.specifier: user_challenge for user_challenge in user_challenges
+    }
+    return [specifier_map[s] for s in specifiers if s in specifier_map]
```

Comment:
isn't the dict non-deterministic in sort order?

Reply:
maybe, but the `specifiers` is in correct order, so it'll return in correct order